zulipterminal/ui_tools/buttons.py

```diff
@@ -53,21 +59,22 @@ def update_count(self, count: int) -> None:
     def update_widget(self, count_text: str) -> Any:
         # Note that we don't modify self._caption
         max_caption_length = (self.width_for_text_and_count -
-                              len(count_text))
+                              len(count_text) - len(self.postfix_text))
         if len(self._caption) > max_caption_length:
             caption = (self._caption[:max_caption_length-1] +
                        '\N{HORIZONTAL ELLIPSIS}')
         else:
             caption = self._caption
         num_extra_spaces = (
             self.width_for_text_and_count - len(count_text) - len(caption)
+            - len(self.postfix_text)
```

Comment:
Do we actually need to keep track of `self.postfix_text`? Can we include it in another variable? This would make this cleaner.

Reply:
Since `self.postfix_caption` is a of type `Union[str, Tuple[Any, str]]`. Determining the length would require me to check instance. To avoid checking instance within the `update_widget` method, I had to keep track of `self.postfix_text`. Another approach would be to modify the parameters taken in by `update_count` and `update_widget`, or rather introducing an `update_postfix` method, which I thought might be a little messy. So, what exactly do I do here?

Reply:
Or I could replace `self.postfix_text` with a `self.postfix_length`, would that be okay?

Reply:
Well, we definitely need the full (`self.`)`postfix_caption`, for rendering :+1: 

What I'm saying is that the `self.postfix_text` is only used for its length, and it's already known in the `__init__`, so it can be pre-computed. It could be as `self.postfix_length`, but I think it could be simplified further.